### PR TITLE
Frontend service

### PR DESCRIPTION
**Pull request by simrankaur20**
Has a basic nav bar and folder structure of the frontend service.

### CODE REVIEW

File: node/public/JS/bmi-calc.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    } else if (status === 'normal') {
+        return suggestions.NORMAL;
+    } else if (status === 'overweight') {
+        return suggestions.OVERWEIGHT;
+    } else {
+        return suggestions.OBESE;
+    }
+}
+
+function SetClassOfStatus(addclassname, removeclassname) {
```
Comment (cpaliwal) [inline node/public/JS/bmi-calc.js]: first letter in lower case. Also this function seems excessive and very specific to particular id. you can just inline this it's essentially two line remove class then add class. you can call removeClass without checking its existence

File: node/public/JS/bmi-calc.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    if (status.classList.contains(removeclassname)) {
+        status.classList.remove(removeclassname);
+    }
+    if (!status.classList.contains(addclassname)) {
+        status.classList.add(addclassname);
+    }
+
+}
+
+function IsValid(height, weight) {
```
Comment (cpaliwal) [inline node/public/JS/bmi-calc.js]: isValid. Also what happens if there are multiple js (which you can't control) files in a page with same method 'isValid'

File: node/views/bmi-calculator/bmi-calculator.ejs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <input type="text" id="height">
+        <p class="text">Weight</p>
+        <input type="text" id="weight">
+        <p id="result"></p>
+        <p id="status"></p>
+        <p id="suggestion"></p>
+        <button id="calculate-btn">Calculate</button>
+        <p id="enter-details-info">Please enter height [cm] and weight [kg]</p>
+    </div>
+    <script src="./JS/bmi-calc.js"></script>
```
Comment (cpaliwal) [inline node/views/bmi-calculator/bmi-calculator.ejs]: generally paths are lowercase

File: node/public/JS/bmi-calc.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    }
+    document.getElementById("suggestion").classList.add('blue-text')
+
+    var suggestion = getSuggestion(status);
+    document.getElementById("suggestion").innerText = ' Suggestion : ' + suggestion;
+
+
+}
+var calculatebtn = document.getElementById('calculate-btn');
```
Comment (cpaliwal) [inline node/public/JS/bmi-calc.js]: calculateBtn

File: node/public/JS/bmi-calc.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+}
+const suggestions = {
+    UNDERWEIGHT: 'Eat more frequently.Choose nutrient-rich foods. Drink water regulary and avoid skipping meals.',
+    NORMAL: 'Eat meals regulary and exercise',
+    OVERWEIGHT: 'Exercise daily.Drink good amount of water.Stop eating any kind of junk food.',
+    OBESE: 'Excercise regularly. Make a strict diet plan. It is recommended to consult a dietician.'
+}
+
+function getSuggestion(status) {
+    if (status === 'underweight') {
```
Comment (cpaliwal) [inline node/public/JS/bmi-calc.js]: RHS should be the constants you defined above

File: node/public/JS/bmi-calc.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
+const status = {
+    UNDERWEIGHT: 'underweight',
+    NORMAL: 'normal',
+    OVERWEIGHT: 'overweight',
+    OBESE: 'obese'
+
+}
+
+function getStatus(bmi) {
```
Comment (cpaliwal) [inline node/public/JS/bmi-calc.js]: this appears to be a global function can it interfere with other function in different file with same name

File: node/views/bmi-calculator/bmi-calculator.ejs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
```
Comment (cpaliwal) [inline node/views/bmi-calculator/bmi-calculator.ejs]: Should this file exist independently, will you include this file in home ejs?

File: frontend/index.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    port = 8080
+
+
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.set("view engine", "ejs");
+app.use(express.static(__dirname + '/public'));
+
+app.use("/", homeRoute);
+app.use("/bmi-calculator", bmiRoute);
```
Comment (cpaliwal) [inline frontend/index.js:24]: is this needed?

File: frontend/index.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+var express = require('express'),
+    app = express(),
+    bmiRoute=require("./routes/bmi-calculator");
```
Comment (cpaliwal) [inline frontend/index.js]: should this be separate route?

File: frontend/routes/bmi-calculator.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
+var express = require("express");
+var router  = express.Router();
+
+router.get('/', async function (req, res) {
+     res.render("./bmi-calculator");
```
Comment (vermaankit484) [inline frontend/routes/bmi-calculator.js]: use 2 space indentation in JS

File: frontend/routes/home.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+var express = require("express");
+var router  = express.Router();
```
Comment (vermaankit484) [inline frontend/routes/home.js]: Use const for imports and always prevent using var, use let instead

File: frontend/routes/home.js [hunk 12]
```diff
@@ -0,0 +1,7 @@
...
+
+router.get('/', async function (req, res) {
+     res.render("home");  
+});
```
Comment (vermaankit484) [inline frontend/routes/home.js]: try to use arrow functions

File: node/public/JS/bmi-calc.js (hunk 1 above)
Comment (simrankaur20) [reply node/public/JS/bmi-calc.js]: okay

File: node/views/bmi-calculator/bmi-calculator.ejs (hunk 3 above)
Comment (simrankaur20) [reply node/views/bmi-calculator/bmi-calculator.ejs]: okay

File: node/public/JS/bmi-calc.js (hunk 4 above)
Comment (simrankaur20) [reply node/public/JS/bmi-calc.js]: okay

File: node/public/JS/bmi-calc.js (hunk 5 above)
Comment (simrankaur20) [reply node/public/JS/bmi-calc.js]: done

File: node/views/bmi-calculator/bmi-calculator.ejs (hunk 7 above)
Comment (simrankaur20) [reply node/views/bmi-calculator/bmi-calculator.ejs]: changed

File: frontend/index.js (hunk 9 above)
Comment (simrankaur20) [reply frontend/index.js]: I have added bmi-calculator in navbar that is a separate page. Hence u have used this route to fire get request and render bmi-calculator ejs page.

File: frontend/index.js (hunk 8 above)
Comment (simrankaur20) [reply frontend/index.js:24]: To render the ejs page, I would perhaps need this route.

File: frontend/routes/home.js (hunk 11 above)
Comment (simrankaur20) [reply frontend/routes/home.js]: done

File: frontend/routes/home.js (hunk 12 above)
Comment (simrankaur20) [reply frontend/routes/home.js]: okay

File: frontend/routes/bmi-calculator.js (hunk 10 above)
Comment (simrankaur20) [reply frontend/routes/bmi-calculator.js]: okay

**Review by cpaliwal — APPROVED** [review]

**Pull request merged**